### DISCUSSION TITLE
comment for future redesign of the matlab diagnostics.

### DISCUSSION THREAD

**Discussion by hkershaw-brown** [NCAR/DART#357] [Ideas]
this is a comment for future redesign of the matlab diagnostics.  there's something not right about how this was implemented if adding a new model to the diagnostics requires every .m file to have the case statement updated with the name of the model.  

for example, the pinfo structure could define categories of models (e.g. 1d, 3d, etc), in addition to the model name.  to add a new model that is similar to an existing one, you set the appropriate category string.  then these files have a case statement on pinfo.category.  as long as you're not adding a model that needs different code, you wouldn't have to change the .m files at all.

_Originally posted by @nancycollins in https://github.com/NCAR/DART/pull/262#discussion_r765087697_

**Comment by kdraeder** [comment]
If pyDARTdiags will replace the Matlab diagnostics, this discussion could be closed,
or the ideas applied to that project.

**Comment by hkershaw-brown** [comment]
boom! closed.
